torch/onnx/utils.py

```diff
@@ -484,63 +500,84 @@ def forward(self, x):
         :class:`torch.onnx.errors.OnnxExporterError`: Other errors that can occur during export.
             All errors are subclasses of :class:`errors.OnnxExporterError`.
     """
+    if operator_export_type != _C_onnx.OperatorExportTypes.ONNX:
+        warnings.warn(
+            "Setting `operator_export_type` to something other than default is deprecated. "
+            "The option will be removed in a future release.",
+            category=FutureWarning,
+        )
+    if training != _C_onnx.TrainingMode.EVAL:
+        warnings.warn(
+            "Setting `training` to something other than default is deprecated. "
+            "The option will be removed in a future release. Please set the training mode "
+            "before exporting the model.",
+            category=FutureWarning,
+        )
+
+    args = (args,) if isinstance(args, torch.Tensor) else args
 
     if dynamo:
         if isinstance(model, (torch.jit.ScriptModule, torch.jit.ScriptFunction)):
             raise TypeError(
                 "Dynamo export does not support ScriptModule or ScriptFunction."
             )
-        # Unsupported parameters for dynamo export
-        # TODO: These are not supported AT THE TIME
+        # TODO(justinchuby): Remove the warning once logic migration is done
         warnings.warn(
-            "f, export_params, verbose, training, input_names, output_names, operator_export_type, opset_version, "
+            "export_params, verbose, training, input_names, output_names, operator_export_type, opset_version, "
             "do_constant_folding, keep_initializers_as_inputs, custom_opsets, export_modules_as_functions, and "
             "autograd_inlining are not supported for dynamo export at the moment."
         )
-        args = _decide_input_format(model, args)
-        kwargs = {}
-        if args is not None and isinstance(args[-1], dict):
-            kwargs = args[-1]
-            args = args[:-1]
-        # TODO: refactor this when we have migrated ExportedProgam and
-        # needs users to specify dynamic_axes
-        dynamic_shapes = _from_dynamic_axes_to_dynamic_shapes(
-            model, dynamic_axes, input_names
-        )
-        exported_program = torch.export.export(
-            model, args=args, kwargs=kwargs, dynamic_shapes=dynamic_shapes  # type: ignore[arg-type]
-        )
-        # TODO: expose ExportOptions?
+        if isinstance(model, torch.export.ExportedProgram):
+            # We the model is already exported program, so the args, kwargs, and dynamic_shapes
+            # are not used
+            args = ()
```

Comment:
Does this make sense? When exported program exists, we don't need args and kwargs to dynamo_export?

Reply:
The args are not used so

Reply:
But we can also preserve them

Reply:
Oh I see. I guess preserving them for debugging would be better choice?

Reply:
Will do

Reply:
Done